ompi/mca/common/ompio/common_ompio_buffer.c

```diff
@@ -67,13 +62,15 @@ static void* mca_common_ompio_buffer_alloc_seg ( void*ctx, size_t *size )
 {
     char *buf=NULL;
     size_t realsize, numpages;
+    int tmp = -1;
+    uint64_t flags = 0;
 
     numpages = (*size + mca_common_ompio_pagesize -1 )/mca_common_ompio_pagesize;
     realsize = numpages * mca_common_ompio_pagesize;
 
     buf = malloc ( realsize);
 
-    if (NULL != buf && NULL != opal_accelerator.host_register) {
+    if (NULL != buf && opal_accelerator.check_addr(buf, &tmp, &flags) == 0) {
```

Comment:
Minor NIT, but in OMPI we put constants on the left of == so that we catch errors like writing "=" instead of "==" (assignment errors that can typically not throw an error but will lead to some very bad behavior

Reply:
fixed everywhere